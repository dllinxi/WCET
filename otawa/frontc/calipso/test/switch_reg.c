
void main(void) {
	char c;
	int i;
	
	/* very simple */
	switch(c) {
	case 'a':
		i++;
		break;
	case 'b':
		i = 0;
		break;
	case 'c':
		i--;
		break;
	default:
		i = 0;
	}
	
	/* a bit more complex */
	switch(c) {
	case 'a':
		i++;
		break;
	case 'b': {
			if(i == 0)
				i++;
			else
				i = 0;
			break;
		}
	case 'c':
		if(i == 0) {
			i++;
			break;
		}
		else {
			i--;
			break;
		}
	default:
		i = 0;
	}
	
}
