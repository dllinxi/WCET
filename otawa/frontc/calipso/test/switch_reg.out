
void main(void )
{
	char c;
	int i;
	switch(c)
	{
	case 'a':
			i++;
		break;
	case 'b':
			i = 0;
		break;
	case 'c':
			i--;
		break;
	default :
			i = 0;
	}
	switch(c)
	{
	case 'a':
			i++;
		break;
	case 'b':
		{
			if(i == 0)
				i++;
			else
				i = 0;
			break;
		}
	case 'c':
		{
			if(i == 0)
			{
				i++;
				break;
			}
			else
			{
				i--;
				break;
			}
		}
	default :
			i = 0;
	}
}

